src/lib/unpackPlotProps.js

```diff
@@ -1,9 +1,102 @@
+/* eslint-disable no-console */
+
 import nestedProperty from 'plotly.js/src/lib/nested_property';
 import isNumeric from 'fast-isnumeric';
 import {MULTI_VALUED, MULTI_VALUED_PLACEHOLDER} from './constants';
 
+const hasFullValue = fullValue => fullValue !== void 0 && fullValue !== null;
+
+export function isValidCustomConfigObject(customConfig) {
+  if (
+    customConfig &&
+    customConfig === Object(customConfig) &&
+    Object.keys(customConfig).length &&
+    customConfig.visibility_rules
+  ) {
+    const logRuleError = () => {
+      console.error(
+        'invalid customConfig: all customConfig visibility rules must include type and regex_match keys.'
+      );
+    };
+
+    if (
+      (customConfig.visibility_rules.blacklist &&
+        customConfig.visibility_rules.blacklist.some(r => !r.type || !r.regex_match)) ||
+      (customConfig.visibility_rules.whitelist &&
+        customConfig.visibility_rules.whitelist.some(r => !r.type || !r.regex_match))
+    ) {
+      logRuleError();
+      return false;
+    }
+
+    if (
+      customConfig.visibility_rules.order &&
+      Array.isArray(customConfig.visibility_rules.order) &&
+      customConfig.visibility_rules.order.every(o => ['blacklist', 'whitelist'].includes(o))
+    ) {
+      console.error(
+        "invalid customConfig: only 2 values are accepted in the visibility_rules.order array: 'blacklist' and 'whitelist'"
+      );
+      return false;
+    }
+
+    return true;
+  }
+  return false;
+}
+
+export function applyCustomConfigVisibility(props, fullValue, customConfig) {
```

Comment:
consider renaming to `isVisibleAccordingToCustomVisibilityRules` or soemthing shorter

Reply:
I would remove the `hasFullValue` checks from inside this function, so this is a "permitted but not enforced" function